macros/edr/materializations/test/test.sql

```diff
@@ -133,6 +137,77 @@
   {% do return(elementary.agate_to_dicts(elementary.run_query(query))) %}
 {% endmacro %}
 
+{% macro get_columns_to_exclude_from_sampling(flattened_test) %}
+  {% set columns_to_exclude = [] %}
+  
+  {% if not flattened_test %}
+    {% do return(columns_to_exclude) %}
+  {% endif %}
+  
+  {% if elementary.get_config_var('disable_samples_on_pii_tags') %}
+    {% set pii_columns = elementary.get_pii_columns_from_parent_model(flattened_test) %}
+    {% set columns_to_exclude = columns_to_exclude + pii_columns %}
+  {% endif %}
+  
+  {% if elementary.is_sampling_disabled_for_column(flattened_test) %}
+    {% set test_column_name = elementary.insensitive_get_dict_value(flattened_test, 'test_column_name') %}
+    {% if test_column_name and test_column_name not in columns_to_exclude %}
+      {% do columns_to_exclude.append(test_column_name) %}
+    {% endif %}
+  {% endif %}
+  
+  {% do return(columns_to_exclude) %}
+{% endmacro %}
+
+{# if test query contains PII columns or *, disable sampling entirely #}
+{% macro should_disable_sampling_for_pii(flattened_test) %}
+  {% if not elementary.get_config_var('disable_samples_on_pii_tags') %}
+    {% do return(false) %}
+  {% endif %}
+  
+  {% set pii_columns = elementary.get_pii_columns_from_parent_model(flattened_test) %}
+  {% if not pii_columns %}
+    {% do return(false) %}
+  {% endif %}
+  
+  {# Get the compiled test query #}
+  {% set test_query = elementary.get_compiled_code(flattened_test) %}
+  {% set test_query_lower = test_query.lower() %}
+  
+  {# Check if query uses * (select all columns) #}
+  {% if '*' in test_query_lower %}
```

Comment:
There is a chance we might over censor here (e.g. if we select * from another table in a CTE), but I guess it's better to be safe.
Maybe just worth adding a comment that clarifies it.